packages/components/src/components/hds/dialog-primitive/header.ts

```diff
@@ -4,17 +4,32 @@
  */
 
 import Component from '@glimmer/component';
+import type { FlightIconSignature } from '@hashicorp/ember-flight-icons/components/flight-icon';
+
+export interface HdsDialogPrimitiveHeaderSignature {
+  Args: {
+    contextualClassPrefix?: string;
+    id?: string;
+    tagline?: string;
+    icon?: FlightIconSignature['Args']['name'] | false;
```

Comment:
I _think_ you don't need the `false`: if the argument is not passed is `undefined` which is _falsy_, which gets interpreted by `{{#if @icon}}`. Right @alex-ju ?
```suggestion
    icon?: FlightIconSignature['Args']['name'];
```

Reply:
Ahh, makes sense. I'll update.